### ISSUE TITLE
overlapping trees for annotation macro when code coverage is turned on

### ISSUE THREAD

**Issue by swachter** [scalameta/scalameta#780]
A test project to reproduce can be found here: https://github.com/swachter/macro-scoverage

Compiling the macro annotation

```
package test

import scala.annotation.StaticAnnotation
import scala.meta._

class MacroAnnotation extends StaticAnnotation {

    inline def apply(defn: Any): Any = meta {
      defn match {
        case q"trait $name extends $something" =>
          println("############################")
          q"trait $name extends $something"
        case _ =>
          abort("@MacroAnnotation must annotate a trait")
      }
    }
}
```
with code coverage turned on using the command

```
sbt coverage compile 
```
yields an overlapping trees error:

```
[info] ======= Position error
[info] Overlapping trees (1050,1061)
[info] == Ancestor tree [1062] of type PackageDef at [0:426]MacroAnnotation.scala
[info] 
[info] [L   1        ] #1062   [0:426]         PackageDef // test
[info] 
[info] == First overlapping tree [1050] of type ClassDef at [76:426]MacroAnnotation.scala
[info] 
[info] [L   6        ] #1050   [76:426]        ClassDef   // MacroAnnotation extends StaticAnnotation {
[info] 
[info] == Second overlapping tree [1061] of type ModuleDef at [76:426]MacroAnnotation.scala
[info] 
[info] [L   6        ] #1061   [76:426]        ModuleDef  // MacroAnnotation extends StaticAnnotation {
[info] 
[info] 
[info] While validating #1062
[info] [L   1        ] #1062   [0:426]         PackageDef // test
[info] 
[info] Children:
[info]   [L   1 P# 1062] #23     [8:12]          Ident      // test
[info]   [L   3 P# 1062] #26     [14:54]         Import     // scala.annotation.StaticAnnotation
[info]   [L   4 P# 1062] #29     [55:74]         Import     // scala.meta._
[info]   [L   6 P# 1062] #1050   [76:426]        ClassDef   // MacroAnnotation extends StaticAnnotation {
[info]   [L   6 P# 1062] #1061   [76:426]        ModuleDef  // MacroAnnotation extends StaticAnnotation {
[info] =======
```
Note: I do not know if this error is related to macros or to code coverage.

**Comment by olafurpg** [comment]
Thank you for reporting! We merged this week a workaround for this issue https://github.com/scalameta/paradise/pull/183 You can try a pre-release version `3.0.0-184` 
```
resolvers += Resolver.bintrayIvyRepo("scalameta", "maven")
```
Please report back if that fixes the issue

**Comment by swachter** [comment]
Thanks for the quick response!

I tried the pre-release (and pushed the changes to GitHub). Now I get a compilation error:

```
MacroAnnotation.scala:8: error: ; expected but def found
inline def apply(defn: Any): Any = meta {
           ^
	at scala.meta.internal.parsers.Reporter.syntaxError(Reporter.scala:17)
	at scala.meta.internal.parsers.Reporter.syntaxError$(Reporter.scala:17)
	at scala.meta.internal.parsers.Reporter$$anon$1.syntaxError(Reporter.scala:23)
```

**Comment by olafurpg** [comment]
Hmmm, I will take a look, thanks!

**Comment by olafurpg** [comment]
I updated the meta dependency in paradise https://github.com/scalameta/paradise/pull/185 and that seems to fix the issue in macro-scoverage (tested with publishLocal).

Once that PR is merged you can try it out with 3.0.0-185, but note that you need to also upgrade the scala.meta dependency like here https://github.com/swachter/macro-scoverage/commit/97367ba8df41cb44952a30dbb43b148ef7d17eb1#diff-fdc3abdfd754eeb24090dbd90aeec2ceR7
